src/Pim/Bundle/CatalogBundle/Doctrine/Common/Saver/ProductSaver.php

```diff
@@ -115,4 +112,50 @@ protected function validateProduct($product)
             );
         }
     }
+
+    /**
+     * Calculates current product completenesses.
+     *
+     * The current completenesses collection is first cleared, then newly calculated ones are set to the product.
+     *
+     * @param ProductInterface $product
+     */
+    protected function calculateProductCompletenesses(ProductInterface $product)
```

Comment:
please, take an array of products as input here
and perform only one query to drop all the completenesses of the bulk